[BUG] Issue building Dockerfile in Chariott

<!--- Provide a general summary of the issue in the Title above -->
There is currently an issue building Chariott when building a Dockerfile. This issue is not reproducible when directly running the cargo build command for Chariott.

## Expected Behavior
<!--- What should happen -->
First, Dockerfiles in the repo should be able to be built without errors. Secondly, if there is an issue building the code, it should be reproducible both in the Docker build and when directly building the code with cargo.

## Current Behavior
<!--- What is actually happening -->
Chariott can be built directly with cargo without any issues. However, when a Dockerfile is attempted to be built, using the same cargo command, the build fails with an error.

## Steps to Reproduce
<!-- Provide a detailed set of steps to reproduce the bug. -->

1. run `rustup target add x86_64-unknown-linux-musl` to add target.
2. run `cargo build --release --target=x86_64-unknown-linux-musl` in the root of the repo, and see that it completes without error.
3. then, attempt to build a Dockerfile with `docker build -t chariott -f ./Dockerfile . ` in the root of the repo. This should result in an error, of type `error[E0658]: attributes on expressions are experimental`.

## Context (Environment)

<!-- Provide some context to the bug, e.g. on which platform, with which account, ... -->
Environment: WSL 2
cargo version: 1.65.0 (4bc8f24d3 2022-10-20)
Docker version: 23.0.1, build a5ee5b1
rust compiler (rustc) version: 1.65.0 (897e37553 2022-11-02) 

## Logs
<!-- Please share as many logs as possible. This will help to debug -->
<!-- If logs are long, considering putting them in a [gist](https://gist.github.com/) and adding the link here. This helps keep this issue report smaller. -->
<!-- Don't forget to remove any connection string information! -->

> 154.0    Compiling chariott-common v0.1.0 (/sdv/common)
> 154.7 error[E0658]: attributes on expressions are experimental
> 154.7       --> src/chariott_grpc.rs:144:9
> 154.7        |
> 154.7 144 |         #[cfg(test)]
> 154.7        |         ^^^^^^^^^^^^
> 154.7        |
> 154.7        = note: see issue #15701 <https://github.com/rust-lang/rust/issues/15701> for more information
> 154.7 
> 154.7 error: removing an expression is not supported in this position
> 154.7       --> src/chariott_grpc.rs:144:9
> 154.7        |
> 154.7 144 |         #[cfg(test)]
> 154.7        |         ^^^^^^^^^^^^
> 154.7 
> 155.0 For more information about this error, try `rustc --explain E0658`.
> 155.0 error: could not compile `chariott` due to 2 previous errors


## Additional Information
<!-- Please provide any additional information that may be helpful in understanding the issue. -->

--- a/.github/workflows/containerize.yaml
+++ b/.github/workflows/containerize.yaml
@@ -41,7 +41,7 @@ jobs:
 
       - name: Find Rust version
         id: rust_version
-        run: echo "::set-output name=RUST_VERSION::$(awk -F "=" '/channel/ {print $2}' rust-toolchain.toml)"
+        run: echo "::set-output name=RUST_VERSION::$(awk -F "=" '/channel/ {gsub(/ /, "", $2); gsub(/"/, "", $2); print $2}' rust-toolchain.toml)"
 
       - name: Set up Docker buildx
         id: buildx

--- a/src/chariott_grpc.rs
+++ b/src/chariott_grpc.rs
@@ -142,7 +142,9 @@ impl<T: Observer + Send + Sync + 'static> ChariottService for ChariottServer<T> 
         #[cfg(not(test))]
         let broker = &self.broker;
         #[cfg(test)]
-        let broker = tests::MockBroker::new(self.broker.clone());
+        let _ = self.broker; // Suppress dead code warning when test feature is active.
+        #[cfg(test)]
+        let broker = tests::MockBroker;
 
         let binding =
             broker.resolve(&config).ok_or_else(|| Status::not_found("No provider found."))?;
@@ -404,11 +406,6 @@ mod tests {
 
     impl MockBroker {
         const RETURN_VALUE: i32 = 10;
-
-        pub fn new(broker: IntentBroker) -> Self {
-            _ = broker; // Suppress dead code warning when test feature is active.
-            Self
-        }
 
         pub fn resolve(&self, _: &IntentConfiguration) -> Option<RuntimeBinding<GrpcProvider>> {
             Some(RuntimeBinding::Test(TestBinding::new(

--- a/src/chariott_grpc.rs
+++ b/src/chariott_grpc.rs
@@ -142,9 +142,7 @@ impl<T: Observer + Send + Sync + 'static> ChariottService for ChariottServer<T> 
         #[cfg(not(test))]
         let broker = &self.broker;
         #[cfg(test)]
-        _ = self.broker; // Suppress dead code warning when test feature is active.
-        #[cfg(test)]
-        let broker = tests::MockBroker;
+        let broker = tests::MockBroker::new(self.broker.clone());
 
         let binding =
             broker.resolve(&config).ok_or_else(|| Status::not_found("No provider found."))?;
@@ -406,6 +404,11 @@ mod tests {
 
     impl MockBroker {
         const RETURN_VALUE: i32 = 10;
+
+        pub fn new(broker: IntentBroker) -> Self {
+            _ = broker; // Suppress dead code warning when test feature is active.
+            Self
+        }
 
         pub fn resolve(&self, _: &IntentConfiguration) -> Option<RuntimeBinding<GrpcProvider>> {
             Some(RuntimeBinding::Test(TestBinding::new(

